akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -71,7 +105,7 @@ private[stream] class Throttle[T](cost: Int,
     override protected def onTimer(key: Any): Unit = {
       push(out, currentElement.get)
       currentElement = None
-      lastTokens = 0
+      recordedTokens = 0 // This will be recalculated automatically once the next onPush is invoked
```

Comment:
This should be moved upwards, next to the modification of the last recorded timestamp, to keep related things together.